app/code/Ometria/Core/Helper/Cookiechannel.php

```diff
@@ -1,97 +1,185 @@
 <?php
-namespace Ometria\Core\Helper; 
-use Magento\Framework\App\Helper\AbstractHelper; 
-use Magento\Framework\App\Helper\Context; 
-class Cookiechannel extends AbstractHelper 
-{
+namespace Ometria\Core\Helper;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Ometria\Core\Helper\Is\Frontend as IsFrontend;
 
-    const COOKIE_NAME = 'ommage';
+class Cookiechannel extends AbstractHelper
+{
+    const COOKIE_NAME                = 'ommage';
+    const COOKIEBOT_COOKIE_NAME      = 'CookieConsent';
     const SEPERATOR_BETWEEN_COMMANDS = ';';
-    const SEPERATOR_IN_COMMANDS = ':';
-
-    protected $helperConfig;
-    
-    protected $frontendAreaChecker;       
-    
-    protected $cookie_did_change = false;        
-    
+    const SEPERATOR_IN_COMMANDS      = ':';
+
+    /** @var Config */
+    private $helperConfig;
+
+    /** @var IsFrontend */
+    private $frontendAreaChecker;
+
+    /** @var bool */
+    private $cookieDidChange = false;
+
+    /**
+     * @param Context $context
+     * @param Config $helperConfig
+     * @param IsFrontend $frontendAreaChecker
+     */
     public function __construct(
         Context $context,
-        \Ometria\Core\Helper\Config $helperConfig,
-        \Ometria\Core\Helper\Is\Frontend $frontendAreaChecker             
-    )        
-    {
-        $this->helperConfig         = $helperConfig;    
-        $this->frontendAreaChecker  = $frontendAreaChecker;        
+        Config $helperConfig,
+        IsFrontend $frontendAreaChecker
+    ) {
+        $this->helperConfig         = $helperConfig;
+        $this->frontendAreaChecker  = $frontendAreaChecker;
+
         return parent::__construct($context);
     }
-        
-    public function addCommand($command, $replace_if_exists=false, $set_cookie=true){
-        if (!$command || !is_array($command)) return;
 
+    /**
+     * @param $command
+     * @param bool $replaceIfExists
+     */
+    public function addCommand($command, bool $replaceIfExists = false)
+    {
+        if (!$command || !is_array($command)) {
+            return;
+        }
 
         // Return if admin area or API call
-        // if (Mage::app()->getStore()->isAdmin()) return;
-        // if (Mage::getSingleton('api/server')->getAdapter() != null) return;
-        if(!$this->frontendAreaChecker->check())
-        {
+        if(!$this->frontendAreaChecker->check()) {
+            return;
+        }
+
+        if (!$this->helperConfig->isConfigured()) {
             return;
         }
 
-        //$ometria_config_helper = Mage::helper('ometria/config');
-        $ometria_config_helper = $this->helperConfig;
-        if (!$ometria_config_helper->isConfigured()) return;
-        if (!$ometria_config_helper->isUnivarEnabled()) return;
+        if (!$this->helperConfig->isUnivarEnabled()) {
+            return;
+        }
+
+        if ($command[0] == 'identify') {
```

Comment:
I'd say better to use a strict comparison here

Reply:
I agree. Most of the changes here are refactoring the existing legacy code for better readability. In general we do this whenever extending the extension in an existing file with new functionality and aim to change the code as little as possible for obvious reasons. The legacy code from the initial build of the extension is quite messy so this approach is intended to improve things over time with minimal risk of adding breaking changes.

I will switch this to === now though, as it is the correct comparison to make.